pkg/codegen/python/gen.go

```diff
@@ -165,9 +165,6 @@ func (mod *modContext) modNameAndName(pkg *schema.Package, t schema.Type, input
 	}
 
 	modName = tokenToModule(token, pkg, info.ModuleNameOverrides)
-	if modName == mod.mod {
-		modName = ""
-	}
```

Comment:
This looks like it was mistakenly copied from the non-external cases:

`objectType`:

https://github.com/pulumi/pulumi/blob/7772c3e846d028266d4cc056205d64dcca9fbd9d/pkg/codegen/python/gen.go#L217-L219

`tokenToResource`:

https://github.com/pulumi/pulumi/blob/7772c3e846d028266d4cc056205d64dcca9fbd9d/pkg/codegen/python/gen.go#L274-L276